packages/xod-project/src/typeDeduction.js

```diff
@@ -192,26 +193,58 @@ const omitIntermediateAmbiguousPins = (
     )
   );
 
+// :: [a] -> Map a *
+const toMap = R.pipe(R.map(a => [a, true]), R.fromPairs);
```

Comment:
Isn’t it a [`Set`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set)?